durable-storage/benches/avl_tree.rs

```diff
@@ -0,0 +1,163 @@
+    Bytes::from(bytes)
+}
+
+fn get_operations_batch(rng: &mut impl Rng, keys: &[Key], batch_size: usize) -> Vec<Operation> {
+    // let ret = Vec::with_capacity(batch_size);
+    (0..batch_size)
+        .map(|_| {
+            let key: Key = keys
+                .choose(rng)
+                .expect("The keys array is not empty")
+                .clone();
+            match random_range(0..3) {
+                0 => Operation::Get(key),
+                1 => Operation::Upsert(key, get_random_data(rng)),
+                _ => Operation::Delete(key),
+            }
+        })
+        .collect()
+}
+
+/// This bench inserts ~half of the 3 byte key space into a tree.
+/// Then in each operation it does "operations_sample_count"
+/// operations which are evenly split between get, set
+/// and delete.
+fn bench_avl_tree_operations(c: &mut Criterion) {
+    let mut rng = rand::rng();
+    let key_count = 10_000_000;
+    let mut keys = generate_keys(&mut rng, key_count);
+
+    // Creating 5000 samples of 10,000 operations
+    let sample_count = 5000;
+    let operations_per_sample = 10_000;
+    let mut samples: Vec<Vec<Operation>> = Vec::with_capacity(sample_count);
+    for _ in 0..sample_count {
+        samples.push(get_operations_batch(&mut rng, &keys, operations_per_sample));
+    }
+
+    // Setting up the tree
+    let mut tree: Avl = Default::default();
+    keys.truncate(keys.len() / 2);
+    for key in keys.iter() {
+        tree.set(key, get_random_data(&mut rng));
+    }
+
+    c.bench_function("Bench AVL tree with operations", |b| {
+        b.iter(|| {
+            // I have no control on the number of times
+            // criterion runs this function so we need enough
+            // samples so we don't need to copy
+            let operations: Vec<Operation> = samples.pop().unwrap_or_default();
+            for operation in operations {
+                match operation {
+                    Operation::Get(key) => {
+                        tree.get(&key);
+                    }
+                    Operation::Upsert(key, value) => {
+                        tree.set(&key, value);
+                    }
+                    Operation::Delete(key) => {
+                        tree.delete(&key);
+                    }
+                }
+            }
+        })
+    });
+}
+
+fn reference(c: &mut Criterion) {
+    let mut rng = rand::rng();
+    let key_count = 10_000_000;
+    let mut keys = generate_keys(&mut rng, key_count);
+
+    let sample_count = 5000;
+    let operations_per_sample = 10_000;
+    let mut samples: Vec<Vec<Operation>> = Vec::with_capacity(sample_count);
+    for _ in 0..sample_count {
+        samples.push(get_operations_batch(&mut rng, &keys, operations_per_sample));
+    }
+
+    let mut tree: BTreeMap<Key, Bytes> = Default::default();
+    keys.truncate(keys.len() / 2);
+    for key in keys.into_iter() {
+        tree.insert(key, get_random_data(&mut rng));
+    }
```

Comment:
```suggestion
    for key in &keys[..keys.len() / 2] {
        tree.insert(key, get_random_data(&mut rng));
    }
```
Not a huge deal since you're not reusing `keys` but mutating it seems unnecessary.

Reply:
Makes sense, I think the suggestion still applies to `bench_avl_tree_operations` though, where you only need references to the keys.

Reply:
(thought I deleted a duplicate reply to this) I think the suggestion still applies to `bench_avl_tree_operations` since you only need references to the keys.

Reply:
Done. Resolving.